File: 268 - Missing Number/Java/solution.java
class Solution {
    public int missingNumber(int[] nums) {
        int n = nums.length;

        int[] temp = new int[n + 1];

        for (int i = 0; i < n + 1; i++) {
            temp[i] = -1;
        }

        for (int num : nums) {
            temp[num] = 1;
        }

        for (int i = 0; i < n + 1; i++) {
            int num = temp[i];
            if (num == -1) {
                return i;
            }
        }

        return 0;
    }
}

File: 268 - Missing Number/Python/solution.py
class Solution:
    def missingNumber(self, nums: List[int]) -> int:
        n = len(nums)
        temp = [-1] * (n + 1)

        for num in nums:
            temp[num] = 1
        for i, t in enumerate(temp):
            if t == -1:
                return i
        
